src/shared/watchdog.c

```diff
@@ -133,6 +133,8 @@ static int open_watchdog(void) {
         fn = !watchdog_device || path_equal(watchdog_device, "/dev/watchdog") ?
                 "/dev/watchdog0" : watchdog_device;
 
+        watchdog_set_device(fn);
```

Comment:
Do you also need to call `watchdog_free_device()` in the error paths?

Reply:
We only set a new string here if no device has been set before. So removing it would have to be conditional.
And the string is freed anyways in `watchdog_free_device()` later. So it does not leak.
Adding anything here is just extra complexity with no real benefit.